app/rankers/algorithms.py

```diff
@@ -111,6 +112,20 @@ def thompson_sampling(
     return [x[0] for x in scores]
 
 
+async def personalize_topic_slates(input_slate_configs: List['SlateConfigModel'],
+                                   personalized_topics: PersonalizedTopicList) -> List['SlateConfigModel']:
+    """
+    This routine takes a list of slates as input in which must include slates with an associated curator topic
+    label.  It uses the topic_profile that is supplied by RecIt to re-rank the slates according to affinity
+    with items in the user's list.
+    :param input_slate_configs: SlateConfigModel list that includes slates with curatorTopicLabels
+    :param user_id: str indicating user profile for personalization
+    :return: SlateLineupExperimentModel with reordered slates
+    """
+    topic_to_score_map = {t.curator_topic_label: t.score for t in personalized_topics.curator_topics}
+    return sorted(input_slate_configs, key=lambda s: topic_to_score_map.get(s.curator_topic_label, 0), reverse=True)
```

Comment:
Does this mean non-topic slates all get pushed to the bottom of the lineup? Is there anyway for us to say we only want the top performing topic slate rather than all 15?

Reply:
can't we cascade a topN ranker after the personalized ranker?

Reply:
I am also unclear on how it *should* work in the question you raise, but this seems beyond what recommendations-api currently allows, i.e. a lineup within a lineup.  which seems like a bigger design discussion.  i guess one possibility is to assume a lineup with a subset of topic slates and then a param that determines the number of topic slates that should be preserved after personalization.